src/NovaVaultV2.sol

```diff
@@ -31,19 +50,38 @@ contract NovaVaultV2 {
         _;
     }
 
-    function addDex(address _contract) external {
+    modifier onlyAllowedDexContracts() {
+        if (!LibAllowList.contractIsAllowed(msg.sender)) {
+            revert DexContractNotAllowed(msg.sender);
+        }
+        _;
+    }
+
+    constructor(
+        address _sDAI,
+        address _swapFacet,
+        address _owner
```

Comment:
I think we can remove this argument, and use msg.sender as first value for `owner` state variable